app/controllers/measurement_sessions_controller.rb

```diff
@@ -20,25 +20,29 @@ class MeasurementSessionsController < ApplicationController
   layout 'map'
 
   def show
-    @session = MobileSession.find_by_url_token(params[:url_token]) or raise NotFound
-    lat = @session.streams.first.min_latitude.to_f
-    lng = @session.streams.first.min_longitude.to_f
-    stream = @session.streams.first
+    session = Session.find_by_url_token(params[:url_token]) or raise NotFound
+    lat = session.streams.first.min_latitude.to_f
+    lng = session.streams.first.min_longitude.to_f
+    stream = session.streams.first
 
-    map = { zoom:16,lat:lat,lng:lng,mapType:"hybrid" }
-    selected_session_ids = [@session.id]
+    map = { zoom:16, lat:lat, lng:lng }
+    selected_session_ids = [session.id]
     data = {
       heat: { highest: stream.threshold_very_high,
               high: stream.threshold_high,
               mid: stream.threshold_medium,
               low: stream.threshold_low,
               lowest: stream.threshold_very_low },
-      usernames: @session.user.username,
+      usernames: session.user.username,
       sensorId: stream.sensor_id
     }
 
-    redirect_to map_path(
-      :anchor => "/map_sessions?map=#{map.to_json}&selectedSessionIds=#{selected_session_ids.to_json}&data=#{data.to_json}"
-    )
+    anchor = "?data=#{data.to_json}&selectedSessionIds=#{selected_session_ids.to_json}&map=#{map.to_json}"
+
+    if (session.type == "FixedSession")
+      redirect_to fixed_map_url(:anchor => anchor)
+    else
+      redirect_to map_url(:anchor => anchor)
```

Comment:
We don't have a `mobile_map_url`?

Reply:
We do. But `map_url ` seems consistent with `else` branch. If I were to use `mobile_map_url ` I should probably put it into `else if (session.type == "Session")` and then have something else for the default...